core/consensus/aura/primitives/src/lib.rs

```diff
@@ -45,5 +49,70 @@ decl_runtime_apis! {
 
 		// Return the current set of authorities.
 		fn authorities() -> Vec<AuthorityId>;
+
+		/// Construct a call to report the equivocation.
+		fn construct_equivocation_report_call(
+			proof: AuraEquivocationProof<<Block as BlockT>::Header, Signature>,
+		) -> Vec<u8>;
+	}
+}
+
+/// Get slot author for given block along with authorities.
+pub fn slot_author<AuthorityId>(slot_num: u64, authorities: &[AuthorityId]) -> Option<&AuthorityId>
+{
+	if authorities.is_empty() { return None }
```

Comment:
I don't understand why you need these extra checks when the function returns an `Option` and `get` does it for you.

why not just 
```rustc
pub fn slot_author<AuthorityId>(slot_num: u64, authorities: &[AuthorityId]) -> Option<&AuthorityId> {
   authorities.get(slot_num as usize)
}
```